### PR TITLE
Add role definition read, use cloudaccess submodule in install

### PR DESCRIPTION
**Pull request by mrsheepuk**

### CODE REVIEW

**Comment by KashifSaadat** [comment]
You'll need to remove from the time sleep blocks in main.tf

**Comment by mrsheepuk** [comment]
> You'll need to remove from the time sleep blocks in main.tf - in fact you may need something similar in the cloudaccess submodule

Ha, just read this _after_ doing it - yes, indeed, had the same thought!

File: modules/cloudaccess/wf_cluster_manager.tf [hunk 1]
```diff
@@ -13,6 +13,20 @@ resource "azurerm_role_definition" "clustermanager" {
   }
 }
 
+resource "time_sleep" "after_azurerm_role_definition_clustermanager" {
```
Comment (KashifSaadat) [inline modules/cloudaccess/wf_cluster_manager.tf:16]: You could technically just have one of these resources `after_azurerm_role_definition`, with the `depends_on` list including all role definition resources. Even if those role definitions are conditional and may not be created, the `depends_on` will just skip that resource if it's not created so it doesn't cause issues.

Comment (mrsheepuk) [reply modules/cloudaccess/wf_cluster_manager.tf:16]: Yeah, I thought about this - to me it felt cleaner to make things depend only on the things they need, i.e. the assignment depends ONLY on its own role, not all of them. other than the repetition, is there any issue with what's been done here?

Comment (KashifSaadat) [reply modules/cloudaccess/wf_cluster_manager.tf:16]: No issues, I'm fine with that 😊

**Review by KashifSaadat — APPROVED** [review]

**Pull request merged**